### PR TITLE
Delete subnode's machine directories

### LINKED ISSUE #9954: minikube delete doesn't remove subnodes machine directory
```
❯ ls
server-key.pem server.pem
❯ minikube start -p test-profile -n 2
😄  [test-profile] minikube v1.15.1 on Darwin 10.15.7
✨  Automatically selected the docker driver
👍  Starting control plane node test-profile in cluster test-profile
🔥  Creating docker container (CPUs=2, Memory=1988MB) ...
🐳  Preparing Kubernetes v1.20.0 on Docker 20.10.0 ...
🔗  Configuring CNI (Container Networking Interface) ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass

❗  Multi-node clusters are currently experimental and might exhibit unintended behavior.
📘  To track progress on multi-node clusters, see https://github.com/kubernetes/minikube/issues/7538.

👍  Starting node test-profile-m02 in cluster test-profile
🔥  Creating docker container (CPUs=2, Memory=1988MB) ...
🌐  Found network options:
    ▪ NO_PROXY=192.168.49.2
🐳  Preparing Kubernetes v1.20.0 on Docker 20.10.0 ...
    ▪ env NO_PROXY=192.168.49.2
🔎  Verifying Kubernetes components...
🏄  Done! kubectl is now configured to use "test-profile" cluster and "default" namespace by default
❯ ls
server-key.pem   server.pem       test-profile     test-profile-m02
❯ minikube delete -p test-profile
🔥  Deleting "test-profile" in docker ...
🔥  Deleting container "test-profile" ...
🔥  Deleting container "test-profile-m02" ...
🔥  Removing /Users/daehyeok/.minikube/machines/test-profile ...
💀  Removed all traces of the "test-profile" cluster.
❯ ls
server-key.pem   server.pem       test-profile-m02
~/.minikube/machines ❯
```

### PR DESCRIPTION
**Pull request by daehyeok**
fixes #9954

Update delete command to remove all subnode's machine directories.

### CODE REVIEW

File: cmd/minikube/cmd/delete.go [hunk 1]
```diff
@@ -496,6 +497,15 @@ func deleteProfileDirectory(profile string) {
 	}
```
Comment (medyagh) [inline cmd/minikube/cmd/delete.go:497]: thanks for finding out this ! we were not cleaning up the machine folder for multi node


I wonder if there are other places that we are missing this

for example
in func demolish

https://github.com/medyagh/minikube/blob/75cd17801287339a00efe44fbd451b4dbfadef86/pkg/minikube/machine/delete.go#L122

is there a chance we could add this delete Machine directory to wherever it is doing for the single node ?

my undrestanding is, for single node delete the machine folder correctly but for multinode we are not.
is there any other place in the code we could do this ? (as opposed to a deffered clean up?)

**Review by medyagh — CHANGES_REQUESTED** [review]

File: cmd/minikube/cmd/delete.go (hunk 1 above)
Comment (daehyeok) [reply cmd/minikube/cmd/delete.go:497]: Even for the single node, it looks `DeleteHost` not  remove directory.
It's handled by `deleteProfileDirectory`
https://github.com/kubernetes/minikube/blob/f68005724c95b67d4002a0612291c655ec5b921f/cmd/minikube/cmd/delete.go#L330-L331

```
❯ minikube start -p test-profile
😄  [test-profile] minikube v1.16.0 on Darwin 10.15.7
✨  Automatically selected the docker driver
👍  Starting control plane node test-profile in cluster test-profile
🔥  Creating docker container (CPUs=2, Memory=1988MB) ...
🐳  Preparing Kubernetes v1.20.0 on Docker 20.10.0 ...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "test-profile" cluster and "default" namespace by default
❯ minikube delete -p test-profile
🔥  Deleting "test-profile" in docker ...
🔥  Deleting container "test-profile" ...
🔥  Removing /Users/daehyeok/.minikube/machines/test-profile ...
💀  Removed all traces of the "test-profile" cluster.
```

`🔥  Removing /Users/daehyeok/.minikube/machines/test-profile ...` this line printed by `deleteProfileDirectory`
https://github.com/kubernetes/minikube/blob/f68005724c95b67d4002a0612291c655ec5b921f/cmd/minikube/cmd/delete.go#L488-L491

Comment (daehyeok) [reply cmd/minikube/cmd/delete.go:497]: maybe we should fix `delete` function ?
https://github.com/medyagh/minikube/blob/75cd17801287339a00efe44fbd451b4dbfadef86/pkg/minikube/machine/delete.go#L104

Comment (medyagh) [reply cmd/minikube/cmd/delete.go:497]: that is a good idea ! @daehyeok .
admitably our delete code is the messiest and worst part of minikube code (huge huge huge chance for BIG refactor)
once we had an idea to break down the delete code in the CMD package into package.

**Review by medyagh — APPROVED** [review]

**Pull request merged**